### ISSUE TITLE
Theme Editor color value string check fails if not rewritten completely

### ISSUE THREAD

**Issue by BrudiBanani** [florisboard/florisboard#1636]
#### Steps to reproduce
1. Go to Theme Editor
2. Open any rule and edit some color
3. Try to adjust color string, e,g, '#8400abff' to '#8401abff'
4. Value checker will claim it is not valid
5. Delete string and type same string ('#8401abff') into blank field. Value checker has no problem

#### Environment information
- FlorisBoard Version: 0.3.14-rc01
- Install Source: GitHub
- Device: Asus Zenfone 8
- Android: 12

**Comment by patrickgold** [comment]
Thanks for your bug report!

Tried out your steps, even if you skip step 3 the issue is fully reproducible. This points to an internal state issue, I hope that it is on FlorisBoard's side else I will have to wait on the Compose team to fix their library issue. Will investigate now.

**Comment by patrickgold** [comment]
Found the (sneaky) issue using the debugger: In 0.3.14-rc01 I added RTL support. With this, in the UI color codes are forced LTR, but may be embedded in RTL text. To achieve that I use Unicode control characters, the ones I use are those: https://github.com/florisboard/florisboard/blob/97456645e4c0b29afe5eceface776f62007a1734/app/src/main/java/dev/patrickgold/florisboard/common/Unicode.kt#L19-L36

So when you tap to edit the box, the color code you visually see is `#8400abff`, but in reality it is `\u2066#8400abff\u2069` (the `\u` characters are invisible and zero-width). Thus the validator correctly says that the code is not valid. You deleting and reentering the code of course removes the Unicode control characters and thus the validator does not complain anymore. Will provide a fix for this shortly.
